src/web/components/StickyBottomBanner/PuzzlesBanner.tsx

```diff
@@ -0,0 +1,238 @@
+	alreadyVisitedCount: number;
+	engagementBannerLastClosedAt?: string;
+	subscriptionBannerLastClosedAt?: string;
+	weeklyArticleHistory?: WeeklyArticleHistory;
+};
+
+type BuildPayloadProps = BaseProps & {
+	countryCode: string;
+	hasConsentedToArticleCounts: boolean;
+};
+
+type CanShowProps = BaseProps & {
+	asyncCountryCode: Promise<string>;
+	remotePuzzlesBannerConfig: boolean;
+};
+
+// TODO specify return type (need to update client to provide this first)
+const buildPayload = (props: BuildPayloadProps) => {
+	return {
+		tracking: {
+			ophanPageId: window.guardian.config.ophan.pageViewId,
+			platformId: 'GUARDIAN_WEB',
+			clientName: 'dcr',
+			referrerUrl: window.location.origin + window.location.pathname,
+		},
+		targeting: {
+			alreadyVisitedCount: props.alreadyVisitedCount,
+			shouldHideReaderRevenue: props.shouldHideReaderRevenue,
+			isPaidContent: props.isPaidContent,
+			showSupportMessaging: !shouldHideSupportMessaging(props.isSignedIn),
+			engagementBannerLastClosedAt: props.engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt:
+				props.subscriptionBannerLastClosedAt,
+			mvtId: Number(getCookie('GU_mvt_id')),
+			countryCode: props.countryCode,
+			weeklyArticleHistory: getWeeklyArticleHistory(),
+			hasOptedOutOfArticleCount: !props.hasConsentedToArticleCounts,
+		},
+	};
+};
+
+const getPuzzlesBanner = (
+	meta: { [key: string]: any },
+	url: string,
+): Promise<Response> => {
+	const json = JSON.stringify(meta);
+	return fetch(url, {
+		method: 'post',
+		headers: { 'Content-Type': 'application/json' },
+		body: json,
+	});
+};
+
+export const canShow = async ({
+	remotePuzzlesBannerConfig,
+	isSignedIn,
+	asyncCountryCode,
+	contentType,
+	sectionName,
+	shouldHideReaderRevenue,
+	isMinuteArticle,
+	isPaidContent,
+	isSensitive,
+	tags,
+	contributionsServiceUrl,
+	alreadyVisitedCount,
+	engagementBannerLastClosedAt,
+	subscriptionBannerLastClosedAt,
+}: CanShowProps): Promise<CanShowResult> => {
+	const isPuzzlesPage =
+		window.guardian.config.page.section === 'crosswords' ||
+		window.guardian.config.page.series === 'Sudoku';
+
+	if (shouldHideReaderRevenue) {
+		// We never serve Reader Revenue banners in this case
+		return { result: false };
+	}
+
+	if (isPuzzlesPage && remotePuzzlesBannerConfig) {
+		const countryCode = await asyncCountryCode;
+		const hasConsentedToArticleCounts = await getArticleCountConsent();
+		const bannerPayload = buildPayload({
+			isSignedIn,
+			countryCode,
+			contentType,
+			sectionName,
+			shouldHideReaderRevenue,
+			isMinuteArticle,
+			isPaidContent,
+			tags,
+			contributionsServiceUrl,
+			isSensitive,
+			alreadyVisitedCount,
+			engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt,
+			hasConsentedToArticleCounts,
+		});
+		return getPuzzlesBanner(
+			bannerPayload,
+			`${contributionsServiceUrl}/puzzles`,
+		)
+			.then(checkForErrors)
+			.then((response) => response.json())
+			.then((json: { data?: any }) => {
```

Comment:
Can we replace this with `useApi`? There's a lot of logic being duplicated here that's already set there. We're already using other custom hooks here so I think the pattern should be okay?

Reply:
You can't use hooks outside of a function component, unfortunately. It would be good to have a similar utility to be used in non-component contexts like this.

Reply:
Ah, I see. To be honest, I find it a little hard to follow the rendering logic for a banner, I'm more used to everything being declarative in the classic react style, but I can see that this isn't the case here so yes, you're right, a reat hook won't work